pkg/ee/metering/reconcile.go

```diff
@@ -179,50 +176,115 @@ func cleanupOrphanedReportingCronJobs(ctx context.Context, client ctrlruntimecli
 	return nil
 }
 
-// cleanupAllMeteringResources removes all active parts of the metering
-// components, in case the admin disables the feature.
-func cleanupAllMeteringResources(ctx context.Context, client ctrlruntimeclient.Client, meteringConfig *kubermaticv1.MeteringConfiguration) error {
-	key := types.NamespacedName{Namespace: resources.KubermaticNamespace, Name: meteringToolName}
-	if err := cleanupResource(ctx, client, key, &appsv1.Deployment{}); err != nil {
-		return fmt.Errorf("failed to cleanup metering Deployment: %w", err)
+// fetchExistingReportingCronJobs returns a list of all existing reporting cronjobs.
+func fetchExistingReportingCronJobs(ctx context.Context, client ctrlruntimeclient.Client) (*batchv1.CronJobList, error) {
+	existingReportingCronJobs := &batchv1.CronJobList{}
+	listOpts := []ctrlruntimeclient.ListOption{
+		ctrlruntimeclient.InNamespace(meteringNamespace),
+		ctrlruntimeclient.ListOption(ctrlruntimeclient.MatchingLabels{common.ComponentLabel: meteringName}),
+	}
+	if err := client.List(ctx, existingReportingCronJobs, listOpts...); err != nil {
+		if !apierrors.IsNotFound(err) {
```

Comment:
Can a 404 really happen when listing CronJobs?

Reply:
I assume this happens when parallel reconcile is enabled or someone manually deleting it at the same time. However we can always ignore this error.